example_app/app.ts

```diff
@@ -0,0 +1,43 @@
+import { Drash, SocketServer } from "./deps.ts";
+
+import HomeResource from "./src/resources/home_resource.ts";
+import ChatResource from "./src/resources/chat_resource.ts";
+
+const webServer = new Drash.Http.Server({
+  response_output: "application/json",
+  directory: Deno.realPathSync("."),
+  static_paths: ["/public"],
+  resources: [
+    HomeResource,
+    ChatResource,
+  ],
+  template_engine: true,
+  views_path: "./src/views",
+});
+
+webServer.run({
+  hostname: "localhost",
+  port: 3001,
+});
+console.log(`Web server started on ${webServer.hostname}:${webServer.port}`);
+
+const socketServer = new SocketServer();
+socketServer.run({
+  hostname: "localhost",
+  port: 3000,
+});
+console.log(
+  `Socket server started on ${socketServer.hostname}:${socketServer.port}`,
+);
+
+socketServer
+  .createChannel("Channel 1")
+  .onMessage((incomingEvent: any) => {
+    const { message } = incomingEvent;
+    socketServer.getChannel("Channel 1").messages.push({ ...message });
```

Comment:
interesting... i feel like this might be better handled by the server itself. this is going to get overloaded quick / too much responsibility for the Channel. will think about this

Reply:
onMessage comes from the server. createChannel returns the server